docs/BestPractices.md

```diff
@@ -204,3 +205,39 @@ FROM node:alpine as app
 ## Copy built node modules and binaries without including the toolchain
 COPY --from=builder node_modules .
 ```
+
+
+## Smaller images without npm/yarn
+
+If you want to achieve an even smaller image size than the `-alpine`, you can omit the npm/yarn like this:
+
+```Dockerfile
+ARG ALPINE_VERSION=3.16
+
+
+FROM node:18-alpine${ALPINE_VERSION} AS builder
+WORKDIR /build-stage
+COPY package*.json ./
+RUN npm ci
+# Copy the the files you need
+COPY . ./
+RUN npm run build
+
+
+FROM alpine:${ALPINE_VERSION}
```

Comment:
This image won't have the envvar NODE_VERSION like the original Node image does. Is there a way to reliably fix that?

Reply:
It would be nice to have a functionality like this, but unfortunatelly, it does not exists (yet):

```Dockerfile
ENV --from= builder NODE_VERSION =$NODE_VERSION
```

Untile then, the only option that comes up to my mind is to save it to the file and copy the file:

```Dockerfile

RUN echo "$NODE_VERSION" > /tmp/env
...

COPY --from= builder "/tmp/env.json" "/tmp/env"
ENV NODE_VERSION ="$(cat /tmp/env)"
```

For now, I would just skip it.